### PR TITLE
feat: port scheduler sample to python

### PR DESCRIPTION
**Pull request by pashafateev**

### CODE REVIEW

File: samples/scheduler/program.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+from autokitteh.github import github_client
+from autokitteh.slack import slack_client
+
+# Set in "autokitteh.yaml"
+GITHUB_OWNER = os.getenv("GITHUB_OWNER")
+GITHUB_REPO = os.getenv("GITHUB_REPO")
+SLACK_CHANNEL_NAME_OR_ID = os.getenv("SLACK_CHANNEL_NAME_OR_ID")
+
+
+def on_cron_trigger(_):
```
Comment (itayd) [inline samples/scheduler/program.py:25]: @tebeka can you support omitting the argument if not used?

File: samples/scheduler/program.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    """Handles the AutoKitteh cron schedule trigger."""
+
+    github = github_client("github_conn")
+    slack = slack_client("slack_conn")
```
Comment (itayd) [inline samples/scheduler/program.py]: these should be in global scope

File: samples/scheduler/program.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        pr
+        for pr in repo.get_pulls(state="open")
+        if not pr.draft
+        and "draft" not in pr.title.lower()
+        and "wip" not in pr.title.lower()
+    ]
+
+    now = datetime.now(timezone.utc)
+    opened_cutoff = now - timedelta(days=4)
+    update_cutoff = now - timedelta(days=1)
```
Comment (itayd) [inline samples/scheduler/program.py]: the number of days need to be configurable

File: samples/scheduler/program.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    for pr in active_prs:
+        stalled_details = get_stalled_pr_details(pr, now, opened_cutoff, update_cutoff)
+
+        if stalled_details:
+            print(f"PR {pr.number} is stalled")
+            msg += f"\nPR: `{pr.title}`\n  {pr.url}\n  {stalled_details}\n"
+            slack.chat_postMessage(channel=SLACK_CHANNEL_NAME_OR_ID, text=msg)
+
+
+def get_stalled_pr_details(pr, now, opened_cutoff, update_cutoff):
```
Comment (itayd) [inline samples/scheduler/program.py]:
```suggestion
def _get_stalled_pr_details(pr, now, opened_cutoff, update_cutoff):
```

File: samples/scheduler/program.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    ]
+
+    now = datetime.now(timezone.utc)
+    opened_cutoff = now - timedelta(days=4)
+    update_cutoff = now - timedelta(days=1)
+
+    msg = "Daily reminder about stalled PRs:"
+
+    for pr in active_prs:
+        stalled_details = get_stalled_pr_details(pr, now, opened_cutoff, update_cutoff)
```
Comment (itayd) [inline samples/scheduler/program.py]:
```suggestion
        stalled_details = _get_stalled_pr_details(pr, now, opened_cutoff, update_cutoff)
```

File: samples/scheduler/program.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    """Handles the AutoKitteh cron schedule trigger."""
+
+    github = github_client("github_conn")
+    slack = slack_client("slack_conn")
+
+    # Fetch open pull requests that are not drafts or WIP
+    repo = github.get_repo(f"{GITHUB_OWNER}/{GITHUB_REPO}")
+    active_prs = [
+        pr
+        for pr in repo.get_pulls(state="open")
```
Comment (itayd) [inline samples/scheduler/program.py:32]: do we need pagination here?

Comment (pashafateev) [reply samples/scheduler/program.py:32]: No, see here:
https://github.com/PyGithub/PyGithub/blob/c91f26a757f80b893bf1820281ef71986a9af6b2/github/PaginatedList.py#L119-L126

File: samples/scheduler/program.py (hunk 2 above)
Comment (pashafateev) [reply samples/scheduler/program.py]: Done

File: samples/scheduler/program.py (hunk 3 above)
Comment (pashafateev) [reply samples/scheduler/program.py]: Done

File: samples/scheduler/program.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+    if pr.created_at < opened_cutoff:
+        details.append(f"opened {hours_ago(now, pr.created_at)}h ago")
+    if pr.updated_at < update_cutoff:
+        details.append(f"last updated {hours_ago(now, pr.updated_at)}h ago")
+
+    return ", ".join(details)
+
+
+def hours_ago(now, past_time):
```
Comment (itayd) [inline samples/scheduler/program.py]:
```suggestion
def _hours_ago(now, past_time):
```

File: samples/scheduler/program.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def _get_stalled_pr_details(pr, now, opened_cutoff, update_cutoff):
+    """Returns details if a PR is stalled, otherwise returns an empty string."""
+    details = []
+
+    if pr.created_at < opened_cutoff:
+        details.append(f"opened {hours_ago(now, pr.created_at)}h ago")
+    if pr.updated_at < update_cutoff:
+        details.append(f"last updated {hours_ago(now, pr.updated_at)}h ago")
```
Comment (itayd) [inline samples/scheduler/program.py]:
```suggestion
    if pr.created_at < opened_cutoff:
        details.append(f"opened {_hours_ago(now, pr.created_at)}h ago")
    if pr.updated_at < update_cutoff:
        details.append(f"last updated {_hours_ago(now, pr.updated_at)}h ago")
```

**Review by itayd — APPROVED** [review]

**Pull request merged**